src/components/TimerTracker/TimerTracker.tsx

```diff
@@ -0,0 +1,386 @@
+                case 'xy': {
+                    const xyRounds = timer.rounds ?? 0;
+                    const roundDuration = ((timer.roundMinutes ?? 0) * 60 + (timer.roundSeconds ?? 0)) * 1000;
+                    return total + xyRounds * roundDuration;
+                }
+                default:
+                    return total;
+            }
+        }, 0);
+
+        const totalElapsedMilliseconds = completedMilliseconds + elapsedMilliseconds;
+
+        return (totalElapsedMilliseconds / totalMilliseconds) * 100;
+    }, [timerSequence, currentTimerIndex, elapsedMilliseconds, totalMilliseconds]);
+
+    const TimerItem = ({ timer, index }: { timer: TimerSequenceItem; index: number }) => {
+        const [, dragRef] = useDrag({
+            type: ItemType.TIMER,
+            item: { index },
+        });
+
+        const [, dropRef] = useDrop({
+            accept: ItemType.TIMER,
+            hover: (draggedItem: { index: number }) => {
+                if (draggedItem.index !== index) {
+                    onReorderTimers(draggedItem.index, index);
+                    draggedItem.index = index;
+                }
+            },
+        });
+
+        return (
+            <div
+                ref={(node) => dragRef(dropRef(node))}
+                className={`${styles.timerSquare} ${
+                    index < currentTimerIndex ? styles.completed : ''
+                } ${index === currentTimerIndex ? styles.active : ''}`}
+                onClick={() => onTimerSelect(index)}
+            >
+                <div className={styles.completeIndicator}>
+                    <Icon iconName="checkmark" classes={styles.completedMark} strokedClasses={styles.completedStroke} />
+                </div>
+                <div className={styles.incomplete}>
+                    <div className={styles.indicator}>
+                        <h2>{timer.label ?? ''}</h2>
+                        <div className={styles.activeIndicator} />
+                    </div>
+                    <div className={styles.labelArea}>
+                        <h2>{formatTimerNumber(index + 1)}</h2>
+                        {timer.type && (
+                            <Icon
+                                iconName={timer.type}
+                                classes={styles.iconTrack}
+                                filledClasses={styles.filled}
+                                filledHighlightsClasses={styles.filled}
+                                strokedClasses={styles.stroked}
+                                strokedHighlightsClasses={styles.stroked}
+                            />
+                        )}
+                    </div>
+                </div>
+                <TButton
+                    label="Delete"
+                    btnType="small-rect"
+                    actionFunc={() => handleDeleteClick(index)}
+                />
+            </div>
+        );
+    };
+
+    const handleAddTimerClick = () => {
```

Comment:
The share button works great for sharing the config state, but you should also be updating the URL with the new configuration state when you add a timer to the configuration.